api/v1alpha2/conversion.go

```diff
@@ -153,29 +187,42 @@ func (dst *MachineSetList) ConvertFrom(srcRaw conversion.Hub) error {
 
 func (src *MachineDeployment) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*v1alpha3.MachineDeployment)
+	if err := Convert_v1alpha2_MachineDeployment_To_v1alpha3_MachineDeployment(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Manually convert ClusterName from label, if any.
+	if name, ok := src.Labels[MachineClusterLabelName]; ok {
+		dst.Spec.ClusterName = name
+	}
 
 	// Manually restore data.
 	restored := &v1alpha3.MachineDeployment{}
-	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil {
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
 		return err
-	} else if ok {
-		if restored.Spec.Paused {
-			dst.Spec.Paused = restored.Spec.Paused
-		}
 	}
 
-	return Convert_v1alpha2_MachineDeployment_To_v1alpha3_MachineDeployment(src, dst, nil)
+	if restored.Spec.ClusterName != "" {
+		dst.Spec.ClusterName = restored.Spec.ClusterName
```

Comment:
We set dst ClusterName above from the label if it's there. This would overwrite. Do you want to add a comment that this takes precedence?